packages/core/saved-objects/core-saved-objects-migration-server-internal/src/next.ts

```diff
@@ -69,6 +71,12 @@ export const nextActionMap = (client: ElasticsearchClient, transformRawDocs: Tra
       Actions.fetchIndices({ client, indices: [state.currentAlias, state.versionAlias] }),
     WAIT_FOR_YELLOW_SOURCE: (state: WaitForYellowSourceState) =>
       Actions.waitForIndexStatus({ client, index: state.sourceIndex.value, status: 'yellow' }),
+    CHECK_COMPATIBLE_MAPPINGS: (state: CheckCompatibleMappingsState) =>
+      Actions.updateMappings({
```

Comment:
We're effectively updating the index mappings earlier in the flow (if they are actually compatible).
With this change, the `OUTDATED_DOCUMENTS` `_SEARCH` + `_TRANSFORM` might be performed **after** updating the mappings (not the `_meta`).

I believe this shouldn't be a problem cause the mappings are compatible, and we're going to perform an `updateAndPickupMappings` afterwards anyway. WDYT?